bus-mapping/src/evm/opcodes/error_oog_dynamic_memory.rs

```diff
@@ -1,38 +0,0 @@
-use crate::{
-    circuit_input_builder::{CircuitInputStateRef, ExecStep},
-    error::{ExecError, OogError},
-    evm::Opcode,
-    Error,
-};
-use eth_types::{evm_types::OpcodeId, GethExecStep};
-
-#[derive(Clone, Copy, Debug)]
-pub(crate) struct OOGDynamicMemory {}
-
-impl Opcode for OOGDynamicMemory {
-    fn gen_associated_ops(
-        state: &mut CircuitInputStateRef,
-        geth_steps: &[GethExecStep],
-    ) -> Result<Vec<ExecStep>, Error> {
-        let geth_step = &geth_steps[0];
-        debug_assert!([OpcodeId::RETURN, OpcodeId::REVERT].contains(&geth_step.op));
-
-        let mut exec_step = state.new_step(geth_step)?;
-        exec_step.error = Some(ExecError::OutOfGas(OogError::DynamicMemoryExpansion));
-
-        state.stack_read(
-            &mut exec_step,
-            geth_step.stack.last_filled(),
-            geth_step.stack.last()?,
-        )?;
-        state.stack_read(
-            &mut exec_step,
-            geth_step.stack.nth_last_filled(1),
-            geth_step.stack.nth_last(1)?,
-        )?;
-
-        state.handle_return(&mut exec_step, geth_steps, true)?;
```

Comment:
I am not familiar with this code, so just out of curiosity - not sure where is `state.handle_return` called if you use StackOnlyOpcode::gen_associated_ops?

Reply:
Here! https://github.com/scroll-tech/zkevm-circuits/blob/develop/bus-mapping/src/evm/opcodes/stackonlyop.rs#L52

Reply:
Hm, not sure where I was looking at not to see it :). Thanks!